### ISSUE TITLE
Custom 404 Page

### ISSUE THREAD

**Issue by feychou** [redfin/react-server#885]
Is there a way to easily customise the 404 response - for instance with a custom page?

I also tried to add my catch-all route in the route entry file and associate it with a not-found-page: however, this results in a 200. Is there a way to define the 404 status code in the page lifecycle?

**Comment by gigabo** [comment]
> Is there a way to define the 404 status code in the page lifecycle?

Yep, it's not well documented, but you should be able to do this by adding `hasDocument` to your response from `handleRoute`:
```js
class NotFoundPage {
    handleRoute() {
        return {code: 404, hasDocument: true};
    }

    getElements() {
        return <Awesome404 />;
    }
}
```

**Comment by feychou** [comment]
@gigabo That works, thank you. I could open a PR for documenting this somewhere in the docs (maybe where I would have expected to find it). Maybe one of the examples could also have a custom 404 page?

**Comment by gigabo** [comment]
@feychou Yeah, that would be awesome!

**Comment by doug-wade** [comment]
This was fixed in #893; closing
